pulsar-broker/src/main/java/org/apache/pulsar/client/api/RawReader.java

```diff
@@ -49,6 +51,13 @@
      */
     CompletableFuture<RawMessage> readNextAsync();
 
+    /**
+     * Acknowledge all messages as read up until <i>messageId</i>.
+     * @param messageId to cumulatively acknowledge to
+     * @param properties a map of properties which will be stored with the acknowledgement
```

Comment:
it might be worth rephrasing the comment here. acknowledge with properties - does it mean the properties are stored along with individual acknowledgment, or does it mean the properties is part of the subscription.

Reply:
with the individual acknowledgement. will update.